### PR TITLE
prov/efa: Remove deprecated Libfabric APIs in unit tests

### PR DESCRIPTION
**Pull request by jiaxiyan**
FI_ORDER_NONE is removed in OFI 2.0. Removing it is equal to setting it to 0.

FI_MR_BASIC / FI_LOCAL_MR have been deprecated since verison 1.5. The man page recommends that applications migrating from libfabric 1.4 or earlier or wanting to support FI_MR_BASIC set the mr_mode to `FI_MR_VIRT_ADDR | FI_MR_ALLOCATED | FI_MR_PROV_KEY | FI_MR_LOCAL`.

### CODE REVIEW

File: prov/efa/test/efa_unit_test_info.c [hunk 1]
```diff
@@ -77,8 +77,8 @@ void test_info_open_ep_with_api_1_1_info()
 	 * "mode". For example, the requirement of local memory registration
 	 * is specified as FI_LOCAL_MR.
 	 */
-	hints->mode = FI_LOCAL_MR;
-	hints->domain_attr->mr_mode = FI_MR_BASIC;
+	hints->mode = OFI_LOCAL_MR;
```
Comment (shijin-aws) [inline prov/efa/test/efa_unit_test_info.c:80]: We shouldn't use `OFI_*` mode here as `OFI_*` is intended for internal usage without exposure to applications. The blame shows this test was added as part of https://github.com/ofiwg/libfabric/pull/7873 and the PR shows we used to handle `FI_LOCAL_MR` and `FI_MR_BASIC` in the efa provider code specially. It's not clear to me why these mode bits don't show in efa provider now. Can you chase it down?

**Comment by darrylabbate** [comment]
Probably a good idea to capture the required mode bits in a macro, e.g.

```c
#define MR_MODE_BITS FI_MR_VIRT_ADDR | FI_MR_ALLOCATED | FI_MR_PROV_KEY | FI_MR_LOCAL
```

File: prov/efa/test/efa_unit_test_info.c (hunk 1 above)
Comment (jiaxiyan) [reply prov/efa/test/efa_unit_test_info.c:80]: It was removed in https://github.com/ofiwg/libfabric/commit/613e69e9bc46208ce22dd1cdd9f3c59079b6b6a3

Comment (jiaxiyan) [reply prov/efa/test/efa_unit_test_info.c:80]: Do we leave it as it is then?

File: prov/efa/test/efa_unit_test_ep.c [hunk 2]
```diff
@@ -1041,7 +1037,6 @@ void test_efa_rdm_ep_user_zcpy_rx_unhappy_due_to_sas(struct efa_resource **state
 	assert_non_null(resource->hints);
 
 	resource->hints->tx_attr->msg_order = FI_ORDER_SAS;
-	resource->hints->rx_attr->msg_order = FI_ORDER_NONE;
```
Comment (shijin-aws) [inline prov/efa/test/efa_unit_test_ep.c:1044]: tx/rx msg order should be aligned. This was a miss when I added the test. Just make rx attr to be SAS as well here.

**Review by shijin-aws — APPROVED** [review]

**Comment by shijin-aws** [comment]
bot:aws:retest

File: prov/efa/test/efa_unit_test_info.c (hunk 1 above)
Comment (shijin-aws) [reply prov/efa/test/efa_unit_test_info.c]: Just remove it as the tested code path doesn't exist after https://github.com/ofiwg/libfabric/commit/613e69e9bc46208ce22dd1cdd9f3c59079b6b6a3

File: prov/efa/test/efa_unit_tests.h [hunk 3]
```diff
@@ -5,6 +5,9 @@
 #define EFA_UNIT_TESTS_H
 
 #define _GNU_SOURCE
+
+#define MR_MODE_BITS FI_MR_VIRT_ADDR | FI_MR_ALLOCATED | FI_MR_PROV_KEY | FI_MR_LOCAL
```
Comment (darrylabbate) [inline prov/efa/test/efa_unit_tests.h:9]: Let's just make slightly more descriptive, e.g. `MR_BASIC_MODE_BITS`. No strong feelings; `MR_MODE_BITS` is just too vague.

**Review by darrylabbate — APPROVED** [review]

**Pull request merged**